kafka-3-0/pom.xml

```diff
@@ -30,7 +30,7 @@
     <dependency>
       <groupId>com.datastax.oss</groupId>
       <artifactId>kafka-client-api</artifactId>
-      <version>2.10.3.11-SNAPSHOT</version>
+      <version>3.1.2</version>
```

Comment:
this is the version of starlight-for-kafka, not pulsar

should be 3.1.0.1-SNAPSHOT

I think you can set it with the [version plugin](https://www.mojohaus.org/versions/versions-maven-plugin/examples/set.html) 
```
mvn versions:set -DnewVersion=3.1.0.1-SNAPSHOT
```